Src/Base/AMReX_GpuFuse.cpp

```diff
@@ -0,0 +1,241 @@
+        std::memcpy(h_buffer+offset_objects, m_lambda_buf, sizeof_objects);
+        Gpu::htod_memcpy_async(d_buffer, h_buffer, total_buf_size);
+
+        auto d_nwarps = reinterpret_cast<int*>(d_buffer);
+        auto d_lambda_helper = reinterpret_cast<FuseHelper*>(d_buffer+offset_helpers);
+        auto d_lambda_object = reinterpret_cast<char*>(d_buffer+offset_objects);
+
+        constexpr int nthreads = 256;
+        constexpr int nwarps_per_block = nthreads/Gpu::Device::warp_size;
+        int nblocks = (ntotwarps + nwarps_per_block-1) / nwarps_per_block;
+
+        amrex::launch(nblocks, nthreads, Gpu::gpuStream(),
+        [=] AMREX_GPU_DEVICE () noexcept
+        {
+            int g_tid = blockDim.x*blockIdx.x + threadIdx.x;
+            int g_wid = g_tid / Gpu::Device::warp_size;
+            if (g_wid >= ntotwarps) return;
+
+            int ilambda;
+            {
+                int lo = 0;
+                int hi = nlambdas;
+                while (lo <= hi) {
+                    int mid = (lo+hi)/2;
+                    if (g_wid >= d_nwarps[mid] && g_wid < d_nwarps[mid+1]) {
+                        ilambda = mid;
+                        break;
+                    } else if (g_wid < d_nwarps[mid]) {
+                        hi = mid-1;
+                    } else {
+                        lo = mid+1;
+                    }
+                };
+            }
+
+            int b_wid = g_wid - d_nwarps[ilambda]; // b_wid'th warp on this lambda
+            int lane = threadIdx.x % Gpu::Device::warp_size;
+            int icell = b_wid*Gpu::Device::warp_size + lane;
+
+            FuseHelper& helper = d_lambda_helper[ilambda];
+            char* lambda_object = d_lambda_object + helper.m_offset;
+            Box const& bx = helper.m_bx;
+            if (bx.isEmpty()) {
+                if (icell < helper.m_N) {
+                    helper.m_fp.L1D(lambda_object,icell);
+                }
+            } else {
+                int ncells = bx.numPts();
+                if (icell < ncells) {
+                    const auto len = amrex::length(bx);
+                    const auto lo  = amrex::lbound(bx);
+                    int k =  icell /   (len.x*len.y);
+                    int j = (icell - k*(len.x*len.y)) /   len.x;
+                    int i = (icell - k*(len.x*len.y)) - j*len.x;
+                    i += lo.x;
+                    j += lo.y;
+                    k += lo.z;
+                    if (helper.m_N == 0) {
+                        helper.m_fp.L3D(lambda_object,i,j,k);
+                    } else {
+                        for (int n = 0; n < helper.m_N; ++n) {
+                            helper.m_fp.L4D(lambda_object,i,j,k,n);
+                        }
+                    }
+                }
+            }
+        });
+        Gpu::synchronize();
+        The_Pinned_Arena()->free(nwarps);
+        The_Pinned_Arena()->free(h_buffer);
+        The_Device_Arena()->free(d_buffer);
+
```

Comment:
Would it make sense to try to remove this synch and pushback the cleanup into a callback function? Or at least make synching optional, to match the MFIter loop?

Reply:
In the branch that has not been submitted for PR, the `Launch` is called in the destructor of MFIter.  This PR is to get the basic fuse capability into development.  We can worry about how it's integrated into MFIter etc. later.